activejob/lib/active_job/core.rb

```diff
@@ -34,11 +36,30 @@ module Core
 
       # Timezone to be used during the job.
       attr_accessor :timezone
+
+      # Timeout used if job is locking
+      attr_accessor :lock_timeout
+
+      # The key used within the set if the job is locking
+      attr_accessor :lock_key
     end
 
     # These methods will be included into any Active Job object, adding
     # helpers for de/serialization and creation of job instances.
     module ClassMethods
+      attr_accessor :lock_key
+      attr_accessor :lock_timeout
+
+      def lock_key=(lock_key)
+        include Locking unless self <= Locking
```

Comment:
This doesn't seems thread-safe. What is the advantage of conditionally adding this module. This way over asking people to add the module theirselves if they need the feature? Also I don't see why not always include.

Reply:
The advantage of making it conditional is just that it means there's less stuff included in jobs that don't need it. Making it happen when you assign a key or timeout just made it a bit easier as you only have to do those things instead of including it and doing them. I don't have strong opinions either way though and am OK just making it so you have to `include ActiveJob::Locking`.

Reply:
I don't see problem on including things that are not needed dor all jobs. We always add retry and not every job needs it. Even more that those things don't see to be computational heavy.

In Shopify we require users to include the `Locking` module but I guess Rails don't have a lot of APIs that ask users to include modules.